tests/integrationv2/test_record_padding.py

```diff
@@ -0,0 +1,171 @@
+    server_options.mode = Provider.ServerMode
+    server_options.key = certificate.key
+    server_options.extra_flags = None
+
+    s2nd = managed_process(S2N, server_options, timeout=5)
+    openssl = managed_process(provider, client_options, timeout=5)
+
+    for client_results in openssl.get_results():
+        client_results.assert_success()
+        assert_openssl_records_are_padded_correctly(
+            str(client_results.stdout), padding_size)
+
+    expected_version = get_expected_s2n_version(protocol, provider)
+
+    for server_results in s2nd.get_results():
+        server_results.assert_success()
+        # verify that the payload was correctly received by the server
+        assert random_bytes in server_results.stdout
+        # verify that the version was correctly negotiated
+        assert to_bytes("Actual protocol version: {}".format(
+            expected_version)) in server_results.stdout
+        # verify that the cipher was correctly negotiated
+        assert to_bytes("Cipher negotiated: {}".format(
+            cipher.name)) in server_results.stdout
+
+
+@pytest.mark.uncollect_if(func=invalid_test_parameters)
+@pytest.mark.parametrize("cipher", TLS13_CIPHERS, ids=get_parameter_name)
+@pytest.mark.parametrize("provider", [OpenSSL])
+@pytest.mark.parametrize("curve", ALL_TEST_CURVES, ids=get_parameter_name)
+# only tls 1.3 supports record padding
+@pytest.mark.parametrize("protocol", [Protocols.TLS13], ids=get_parameter_name)
+@pytest.mark.parametrize("certificate", MINIMAL_TEST_CERTS, ids=get_parameter_name)
+@pytest.mark.parametrize("padding_size", PADDING_SIZES, ids=get_parameter_name)
+def test_s2n_client_handles_padded_records(managed_process, cipher, provider, curve, protocol, certificate,
+                                           padding_size):
+    port = next(available_ports)
+
+    client_random_bytes = data_bytes(PAYLOAD_SIZE)
+    server_random_bytes = data_bytes(PAYLOAD_SIZE)
+
+    server_options = ProviderOptions(
+        mode=Provider.ServerMode,
+        port=port,
+        cipher=cipher,
+        curve=curve,
+        cert=certificate.cert,
+        key=certificate.key,
+        insecure=True,
+        protocol=protocol,
+        data_to_send=server_random_bytes,
+        extra_flags=['-record_padding', padding_size]
+    )
+
+    client_options = copy.copy(server_options)
+    client_options.mode = Provider.ClientMode
+    client_options.extra_flags = None
+    client_options.data_to_send = client_random_bytes
+
+    # s2nc will wait until it has received the server's response before closing
+    s2nc = managed_process(S2N, client_options, timeout=5,
+                           close_marker=strip_string_of_bytes(str(server_random_bytes)))
+
+    # openssl will send its response after it has received s2nc's record
+    openssl = managed_process(provider, server_options,
+                              timeout=5, send_marker=strip_string_of_bytes(str(client_random_bytes)))
```

Comment:
Is this test succeeding when you run it? I would expect that you need to start the server before the client, or the client will attempt to connect to a nonexistent server and then fail.

Can you post a link to a successful codebuild run of this integration test? You'll have to run it manually.

Reply:
Yes & updated: https://us-west-2.console.aws.amazon.com/codesuite/codebuild/024603541914/projects/S2nIntegrationV2Batch2/batch/S2nIntegrationV2Batch2%3A5f0db7f6-8926-45f9-8e48-1511ef130c0a?region=us-west-2 (S2nIntegrationV2Batch2:5f0db7f6-8926-45f9-8e48-1511ef130c0a)

Reply:
Please comment on the PR with a link like that after every revision. Since the CI doesn't run this test automatically yet, those comments will keep track of what code has been successfully tested.

(A comment on the main PR, not a reply here. It makes the tracking easier)